### ISSUE TITLE
Why need the sign and  multiply it to upper and lower in torch.sigmoid when computing the prior likelihood

### ISSUE THREAD

**Issue by zhengyunqq** [buggyyang/CDC_compression#11]
```python
   def likelihood(self, x, min=1e-9):
        lower = self.cdf(x - 0.5, True)
        upper = self.cdf(x + 0.5, True)
        sign = -torch.sign(lower + upper).detach()
        upper = torch.sigmoid(upper * sign)
        lower = torch.sigmoid(lower * sign)
        return LowerBound.apply(torch.abs(upper - lower), min)
```

**Comment by zzzby-star** [comment]
请问xparam和epsilonparam种做法的区别是什么？作用是什么？

**Comment by zzzby-star** [comment]
What is the difference between xparam and epsilonparam? What is the role?

**Comment by buggyyang** [comment]
@zhengyunqq The implementation was based on the original tensorflow-compression library, I assumed it was designed for better numerical stability so I just rewrote it in pytorch. But I think it's not quite necessary now.

@The-Emperor-0 They are just different parameterizations for diffusion model (check this [paper](https://arxiv.org/abs/2202.00512)). xparameterization has fewer number of evaluation in our case.
